src/store/modules/SearchResults.js

```diff
@@ -103,10 +97,30 @@ export default (fileType) => ({
       // otherwise do api lookup
       commit("setLoading");
       return apiSearch(state.queryString, fileType, batch)
-        .then((newResults) => commit("setResults", { newResults, index: batch * perPage }))
+        .then((newResults) => {
+          const index = batch * perPage;
+          const payload = { newResults, index };
+          commit("setResults", payload);
+          dispatch("fetchNsfw", payload);
+        })
         .catch((error) => {
           commit("setError", { error, batch });
         });
     },
+    async fetchNsfw({ commit }, { newResults, index }) {
+      newResults.hits.forEach((hit, n) => {
+        if (fileType === Types.images) {
```

Comment:
This conditional statement should precede the loop (was already wrong in the original version)

Reply:
pushed a fix